tensorboard/plugins/graph/tf_graph_common/minimap.ts

```diff
@@ -155,7 +155,24 @@ export class Minimap {
     let $download = d3.select('#graphdownload');
     this.download = <HTMLLinkElement>$download.node();
     $download.on('click', d => {
-      this.download.href = this.downloadCanvas.toDataURL('image/png');
+      // Revoke the old URL, if any. Then, generate a new URL.
+      URL.revokeObjectURL(this.download.href);
+      // We can't use the `HTMLCanvasElement.toBlob` API because it does
+      // not have a synchronous variant, and we need to update this href
+      // synchronously. Instead, we create a blob manually from the data
+      // URL.
+      const dataUrl = this.downloadCanvas.toDataURL("image/png");
+      const prefix = dataUrl.slice(0, dataUrl.indexOf(","));
+      if (!prefix.endsWith(":base64") && !prefix.endsWith(";base64")) {
```

Comment:
in which case does the prefix end with ":base64"? maybe the mdn doc is incomplete: https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URIs#Syntax

Reply:
I seem to recall having seen data URIs that do not specify a media type
and are of the form `data:base64,<data>`, even though they really should
be `data:;base64,<data>`. But on further thought I suppose that those
are technically valid data URIs whose media type is `base64` and whose
encoding is textual, not base64, so it would be better to fail here
(which is the more conservative action, anyway). Will patch.